src/main.c

```diff
@@ -1117,6 +1098,71 @@ static int get_root_dir(char *buf, size_t buf_size) {
   }
   setenv("ROOT_DIR", zl_context.root_dir, 1);
   INFO(MSG_ROOT_DIR, buf);
+
+  /* do we really need to change work dir? */
+  if (chdir(zl_context.root_dir)) {
```

Comment:
Doesn't seem to be appropriate to chdir in a function called `get_root_dir`. I suggest to create a separate functions which does `chdir` and sets the env variable.